crates/oxc_linter/src/rules/eslint/valid_typeof.rs

```diff
@@ -31,24 +31,60 @@ pub struct ValidTypeof {
 }
 declare_oxc_lint!(
     /// ### What it does
-    /// Enforce comparing `typeof` expressions against valid strings
+    ///
+    /// Enforce comparing `typeof` expressions against valid strings.
     ///
     /// ### Why is this bad?
-    /// It is usually a typing mistake to compare the result of a `typeof`
-    /// operator to other string literals.
     ///
-    /// ### Example
+    /// For a vast majority of use cases, the result of the `typeof` operator is one of the
+    /// following string literals: `"undefined"`, `"object"`, `"boolean"`, `"number"`, `"string"`,
+    /// `"function"`, `"symbol"`, and `"bigint"`. It is usually a typing mistake to compare the
+    /// result of a `typeof` operator to other string literals.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
     /// ```js
-    /// // requireStringLiterals: false
-    /// // incorrect:
     /// typeof foo === "strnig"
-    /// // correct:
+    /// typeof foo == "undefimed"
+    /// typeof bar != "nunber"
+    /// typeof bar !== "fucntion"
```

Comment:
CI [step](https://github.com/oxc-project/oxc/actions/runs/13875513688/job/38827298292?pr=9797#step:3:41) for the spellchecker is failing. How do I turn off spell check for just these lines without changing the docs?

Reply:
https://github.com/oxc-project/oxc/blob/1d81adac485606bf0636e0d8df58edab728fe5ba/crates/oxc_linter/src/rules/jsdoc/require_param.rs#L91

Reply:
Thanks! @camc314

Unfortunately the inline comment will appear in the docs correctness example. But maybe that is okay.

    `/// typeof bar !== "fucntion"     // spellchecker:disable-line`